File: Hahn.Data/Dtos/Recipies/FoodRecipeDto.cs
using Hahn.Domain.Entities.BaseEntity;
using System.Text.Json.Serialization;

namespace Hahn.Data.Dtos.Recipies;
public class FoodRecipeDto : CommonEntity
{
    [JsonPropertyName("strMeal")]
    public string Title { get; set; }

    [JsonPropertyName("strInstructions")]
    public string Instructions { get; set; }

    [JsonPropertyName("strIngredient1")]
    public string Ingredients { get; set; }

    [JsonPropertyName("strMeasure1")]
    public string Measure { get; set; }
    public string Description { get; set; }
    public string Cuisine { get; set; }
    public string JobId { get; set; }
}

File: Hahn.Frontend/src/types/FoodRecipeDto.ts
// src/types/FoodRecipeDto.ts

export interface Ingredient {
  ingredient: string;
  measure: string;
}

export interface FoodRecipeDto {
  strMeal: string;
  strInstructions: string;
  ingredients: Ingredient[];
  description?: string; // Optional fields
  cuisine?: string;
  jobId?: string | null;
  id: string;
}
